### PR TITLE
[AWQ] Support for a module used in an AWQ mapping to be unquantized && other bug fixes

### PR DESCRIPTION
**Pull request by ZewenShen-Cohere**
This PR addresses the following issues:

1. In some cases, we want to keep specific nn.Modules used in AWQ mappings unquantized to preserve accuracy, such as the MoE gate. This PR introduces a new force_balance field to support this behavior. It also fixes issue [#2151](https://github.com/vllm-project/llm-compressor/issues/2151) and corrects the default MoE AWQ mapping for the post-attention layer norm, which previously did not account for balancing the MoE gate.

2. _accumulate_mean produces incorrect output on its first run.

3. cache_smooth_activations_hook previously performed the averaging computation on the CPU. When both the hidden dimension and sequence length are large, this makes AWQ calibration CPU-bound. The slowdown is especially severe when multiple AWQ quantization jobs run concurrently.

4. Added more informative error logging to the AWQ calibration grid search.

Test Plan

Compare the accuracy of AWQ results produced by the new version against the previous version on the Qwen-3a30t model.

### CODE REVIEW

File: examples/awq/qwen3_moe_example.py [hunk 1]
```diff
@@ -76,9 +84,4 @@ def tokenize(sample):
 )
 output = model.generate(input_ids, max_new_tokens=100)
 print(tokenizer.decode(output[0]))
-print("==========================================\n\n")
```
Comment (ZewenShen-Cohere) [inline examples/awq/qwen3_moe_example.py:79]: I'll revert these changes later

File: src/llmcompressor/modifiers/awq/base.py [hunk 2]
```diff
@@ -588,14 +629,15 @@ def _compute_best_scale(
                 scales[torch.isnan(scales)] = 1
 
                 # Q(W * s)
-                for balance_layer in balance_layers_to_patch:
+                for balance_layer in mapping.balance_layers:
```
[the remark this answers is not in the document]
Comment (ZewenShen-Cohere) [reply src/llmcompressor/modifiers/awq/base.py:632]: Gemini’s answer seems incorrect. We also need to account for nn.Modules that should not be quantized, so that the model produced by AWQ remains functionally equivalent to the original network.

**Comment by HDCharles** [comment]
Hey, I fixed the accumulate means bug already in https://github.com/vllm-project/llm-compressor/pull/2114

the intended behavior is that mappings handle AWQ equalization (scale/rescale) and ignore/targets are for quantization,  you used to be able to set something in mappings and then ignore to equalize it but not quantize it. 

I can see how its valuable to be able to ignore things from the mappings, especially for VL models, but i think the logic doesn't need a new field. I think what should happen is that we should have logic such that first we do a full mapping matching then we compare with the ignore list and if every layer in the set is in the ignore list then we don't do the rescale for it.

**Comment by brian-dellabetta** [comment]
Hi @ZewenShen-Cohere , thanks for the information and suggested changes. I agree with @HDCharles , the `force_balance` field is prone to be confusing and we are looking into a way to resolve this without it.

Your changes listed for numbers 3 and 4 though look good to me, thanks for catching the CPU bottleneck issue. Do you mind modifying this PR (or creating a separate PR) to have just those changes and we can review them separately?

**Comment by ZewenShen-Cohere** [comment]
> Hey, I fixed the accumulate means bug already in #2114

I think the bug that I fixed is different from yours. The bug I fixed is in the calculation of average activation (see [this](https://github.com/vllm-project/llm-compressor/pull/2158/changes#diff-a79ed86c2a25b361e101f22c2cdcc549cf2dbc09b48de91c752f5edc4cf925a7R926))


> I can see how its valuable to be able to ignore things from the mappings, especially for VL models, but i think the logic doesn't need a new field. I think what should happen is that we should have logic such that first we do a full mapping matching then we compare with the ignore list and if every layer in the set is in the ignore list then we don't do the rescale for it.

Thank you for the suggestions. The plan sounds good to me.

**Comment by ZewenShen-Cohere** [comment]
> Your changes listed for numbers 3 and 4 though look good to me, thanks for catching the CPU bottleneck issue. Do you mind modifying this PR (or creating a separate PR) to have just those changes and we can review them separately?

Sure, I'll do that.
